symengine/infinity.cpp

```diff
@@ -238,4 +238,295 @@ inline RCP<const Infty> infty(const RCP<const Number> &direction)
     return make_rcp<Infty>(direction);
 }
 
+class EvaluateInfty : public Evaluate
+{
+    virtual RCP<const Basic> sin(const Basic &x) const override
+    {
+        SYMENGINE_ASSERT(is_a<Infty>(x))
+        const Infty &s = down_cast<const Infty &>(x);
+        if (s.is_positive() or s.is_negative()) {
+            throw DomainError("sin(x) is not defined for x = Infinity");
+        } else {
+            throw DomainError("sin(x) is not defined for x = Complex Infinity");
+        }
+    }
+    virtual RCP<const Basic> cos(const Basic &x) const override
+    {
+        SYMENGINE_ASSERT(is_a<Infty>(x))
+        const Infty &s = down_cast<const Infty &>(x);
+        if (s.is_positive() or s.is_negative()) {
+            throw DomainError("cos(x) is not defined for x = Infinity");
+        } else {
+            throw DomainError("cos(x) is not defined for x = Complex Infinity");
+        }
```

Comment:
Instead of the above 6 lines, do `throw DomainError("cos undefined for infinite values")